sdk/core/azure-core/azure/core/async_paging/_async_page_iterator.py

```diff
@@ -0,0 +1,107 @@
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import collections.abc
+import logging
+from typing import (
+    AsyncIterator,
+    Awaitable,
+    Callable,
+    Iterable,
+    Optional,
+    Tuple,
+)
+
+from ._async_paging_method_handler import _AsyncPagingMethodHandler
+from ._async_list import AsyncList
+from ..paging._utils import ReturnType, ResponseType
+from ..paging import PagingMethodABC
+from ..exceptions import AzureError
+from ..pipeline._tools_async import await_result as _await_result
+
+_LOGGER = logging.getLogger(__name__)
+
+class AsyncPageIterator(AsyncIterator[AsyncIterator[ReturnType]]):
+    def __init__(
+        self,
+        get_next: Optional[Callable[
+            [Optional[str]], Awaitable[ResponseType]
+        ]] = None,
+        extract_data: Optional[Callable[
+            [ResponseType], Awaitable[Tuple[str, AsyncIterator[ReturnType]]]
+        ]] = None,
+        continuation_token: Optional[str] = None,
+        paging_method: Optional[PagingMethodABC] = None,
+        **kwargs
+    ) -> None:
+        """Return an async iterator of pages.
+
+        :param get_next: Callable that take the continuation token and return a HTTP response
+        :param extract_data: Callable that take an HTTP response and return a tuple continuation token,
+         list of ReturnType
+        :param str continuation_token: The continuation token needed by get_next
+        :param paging_method: Preferred way of paging. Pass in a sansio paging method, to tell the iterator
+         how to make requests, and deserialize responses. When passing in paging_method, do not pass in
+         callables for get_next and extract_data.
+        :type paging_method: ~azure.core.paging_method.PagingMethodABC
+        """
+        if get_next or extract_data:
+            if paging_method:
+                raise ValueError(
+                    "You can't pass in both a paging method and a callback for get_next or extract_data. "
+                    "We recommend you only pass in a paging method, since passing in callbacks is legacy."
+                )
+            if not get_next and extract_data:
```

Comment:
I don't think this if statement works - wouldn't it need to be:
`if not (get_next and extract_data):`

Reply:
yep, def right, thanks!